content/blog/2025/anchor-position-area.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Better Anchor Positioning with position-area
+sub: It's not just a shorthand for anchor()
+date: 2025-02-12
+image:
+  src: blog/2025/anchor-grid.jpg
+  alt: >
+    A hand with painted nails
+    placing a white square of paper
+    into a 9 by 9 grid.
+author: james
+sponsors: true
+tags:
+  - Article
+  - Anchor Positioning
+  - CSS
+summary: |
+  `position-area` might be my favorite part of the CSS Anchor Positioning spec,
+  with a ton of features packed in to make things just... work.
+  But there's no magic here, just a few key parts that work well.
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+{% callout 'note', false %}
+
+The examples in this article are best viewed in a Chromium browser, version 131
+or later.
+
+{% endcallout %}
+
+On the surface, `position-area` seems like just a shorthand.
+Just replace `top: anchor(bottom); left: anchor(right)`
+with `position-area: bottom right` and it works.
+
+Or go even further, and replace that with `position-area: end`.
+
+{{ embed.codepen(
+  id='zxOVXOm',
+  title='position-area containing block',
+  user='jamessw',
+  height=300,
+  tab='css,result'
+) }}
+
+The idea behind `position-area` is that in most cases, you want your positioned
+element to be touching a specific corner or edge of the anchor. `position-area`
+creates a grid around the anchor, and lets you place your positioned element
+within that grid.
+
+[Anchor-tool.com](https://anchor-tool.com/) is a fantastic tool for exploring
+the different values for positioning an element within that grid.
+
+But `position-area` is more than a shorthand for `anchor()`, introducing some
+behaviors that may be surprising.
+
+## A new containing block
+
+One of the key differences between using `anchor()` and `position-area` is that
+`position-area` creates a new containing block for the positioned element. While
+we're used to dealing with containing blocks when positioning `position:
+absolute` inside of a `position: relative` ancestor, `position-area` creates a
+containing block that does not map to an element.
+
+This containing block is generated by the grid area that you select using
+`position-area`. For instance, with `position-area: top left`, the new
+containing block starts at the top left corner of the anchor's containing block
+and extends to the top left corner of the anchor itself.
+
+{{ embed.img(
```

Comment:
Having a version of this diagram above for absolute positioning could establish that the entire space is the original containing block - and then this one shows the new containing block.

Reply:
If there's an anchor relationship, there should still be a relationship, right? Or am I missing something? Is the position-area value not always an inset fragment of the containing block that `anchor()` would reference?

Reply:
For positioned elements with absolute positioning, the containing block can be an ancestor of the anchor's containing block. If you're using `anchor()`, the positioned element isn't contained by the anchor's containing block, so there wouldn't be a direct relationship. Or I might be missing your point here?